.travis.yml

```diff
@@ -23,12 +23,17 @@ matrix:
     - php: 7.1
       env:
         - PHPUNIT_TEST=framework
+# Test with lowest composer requirements (i.e. framework 4.0). Ensure that recipe version matches composer.json
+        - COMPOSER_ARG="--prefer-lowest"
+        - RECIPE_VERSION=^4
     - php: 7.2
       env:
         - PHPUNIT_TEST=framework
+        - RECIPE_VERSION=^4.3
```

Comment:
Using a caret here would probably pull recipe 4.6.1, seems like that's not what we want here.  Other travis files use subsequent dev recipes e.g. 4.5.x-dev, 4.6.x-dev, 4.x-dev which is helpful because part of our workflow is to check that travis is green before doing a release, and x-dev being the tip of the branch is what we would release.

The caret does still make sense on the --prefer-lowest entry in the matrix

Reply:
4.x-dev makes sense. 4.6.x-dev will go away at some point so I'll avoid that.

I'll change this one to ~4.3.0

Reply:
I don't understand the reasoning for avoiding recipe 4.6.n though targeting recipe 4.3.n ? - I would have though that recipe 4.6.n is far more relevant?

Reply:
Right. I was just getting a selection of historical versions but you're probably right: oldest, latest stable, and dev are probably the 3 that matter the most.

I want to avoid using 4.6.x-dev, though, as that will break when the 4.6 branch is deleted.

Reply:
Why would we ever delete the 4.6 branch?

If the intention is to lower the amount of toil involved in updating travis files, it's basically unavoidable.  When 4.7.x-dev is released, we're going to have to manually update all the travis files.  That's just the nature of the beast.

From a CI perspective it's more useful to be testing 4.6.x-dev rather than ^4 (which will use 4.6.1) because that way we're more likely to pickup cross-repo regressions caused by recently merged code.  Merge-ups don't always happen (even though they should, humans after all) so I think we do still need 4.6.x-dev as well as 4.x-dev

That's what we've done on pretty much all of the recently updated travis configs